vault/identity_store_oidc.go

```diff
@@ -0,0 +1,743 @@
+package vault
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"time"
+
+	"gopkg.in/square/go-jose.v2"
+
+	uuid "github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/helper/parseutil"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+// types
+type audience []string
+type signingAlgorithm int
```

Comment:
Using a string for signing algorithm would be preferable, and would remove the need for the custom type, iota, etc.